app/services/davenport_computer.rb

```diff
@@ -0,0 +1,54 @@
+require 'davenport'
+
+class DavenportComputer
+  attr_reader :choice
+
+  def initialize(choice)
+    @choice = choice
+  end
+
+  def decode_group(group)
+    if group.length == 1
+      group[0][1]
+    else
+      group.collect do |alt|
+        alt[1]
+      end
+    end
+  end
+
+  def result_as_parto
+    alt_ct = choice.alternatives.count
+    pg = Davenport::PreferenceGraph.new(alt_ct)
+    choice.preferences.each do |pref|
+      pref.extend(Preference::RankCoding)
+      pref_rank = pref.rank_encoding
+      while (pref_rank.length < alt_ct)
+        pref_rank << alt_ct
+      end
```

Comment:
Filling-out the rank derived from preference to the number of alternatives solves a problem in which the preference expression does not include all of the alternatives. This occurs:

1. in the fixtures, which are set-up for select-one. 
1. if additional alternatives are added to the choice after some preferences are expressed (something we currently allow). 

The solution is to place all of the alternatives not in the preference together as equally least preferred. This works for the second case because the new alternatives will have higher id's. It does not work for the first case. What's needed is to fill the rank array with the lowest (highest numbered) rank and then place the ranks of the expressions in the array at the position corresponding to the alternatives they express.

This needs to be revisited only if the site is going to support changing the preference aggregation method after preferences have been expressed. Changing the system after some have already expressed preferences would be quite the mess, although it might be interesting to use the preference expressions to show how the results change when the rules change. For now, this serves.